### PR TITLE
Fix citation cff and add worflow for future validation

### PR DESCRIPTION
**Pull request by Czaki**
# Description

We have a problem that there is no entry on zenodo for napari for 0.4.19 release line. It lokks like some PR (most probably mine) broke the citation cff format. 

I have validated and fixed field based on `cffconvert` python package. I also added GitHub actions for validate file content based on official documentaion https://github.com/citation-file-format/citation-file-format/blob/main/README.md 

I'm not 100% sure that it will be enough to fix it, but it looks required for that.

### CODE REVIEW

**Review by brisvag — APPROVED** [review]

**Comment by jni** [comment]
Thanks @Czaki!

Additionally I would like to reorder the author list — if it has to be automatic then I suggest by number of commits, but I'm open to other suggestions, including e.g. 1st core devs alphabetical then by commits. But all alphabetical is very weird. Personally I think "Sofroniew N, Lambert T, et al" is still a good reference for now.

**Comment by jni** [comment]
(of course, this can come in a different PR.)

**Comment by jni** [comment]
Apparently you can manually edit Zenodo entries so I'll try to get in touch with Nick (who I believe set this up back in the day) to see if we can put credentials/ownership in the joint 1Password account and then we can edit the entry. But let's get the metadata being generated correctly on main first.

**Comment by brisvag** [comment]
Yup I mean after this PR :P I'd say let's merge immediately, not much controversial here.

**Comment by Czaki** [comment]
@jni What are your preferences to above statements?

**Comment by jni** [comment]
I need to think about it, which is why I decided to just merge. 😂

**Comment by Czaki** [comment]
> I need to think about it, which is why I decided to just merge. 😂 

May you open issue with this problem? I think that you could better summarize it and it will be easier to point problem to other core devs.

**Pull request merged**
